bindings/go/oci/semver_tag.go

```diff
@@ -0,0 +1,40 @@
+package oci
+
+import (
+	"strings"
+
+	"ocm.software/open-component-model/bindings/go/oci/internal/log"
+)
+
+// plusSubstitute is used to substitute the plus character ('+') in OCI tags.
+// An OCM version is allowed to contain the plus character, but OCI tags do not allow it.
+// Because the OCI tag of an artifact representing an OCM component is derived from the respective component
+// version, this replacement is required. See also:
+// - https://github.com/open-component-model/ocm-spec/blob/main/doc/04-extensions/03-storage-backends/oci.md#version-mapping
+const (
+	plusSubstitute = ".build-"
+	plus           = "+"
+)
+
+// versionsWithPlus cache is used to keep track of version numbers, a warning has already been logged for.
+// The intention is to avoid logging the same warning multiple times for the same version number.
+var versionsWithPlus = make(map[string]string)
```

Comment:
this is not concurrency safe. if you really want to do this you need to make this safe for loading.

Reply:
I think when we talked about it last time I mentioned this is overkill for now, so id prefer to remove the caching again.